src/crimson/os/seastore/segment_cleaner.h

```diff
@@ -814,10 +858,31 @@ class SegmentCleaner : public SegmentProvider {
       return;
 
     stats.used_bytes += len;
+    auto old_usage = space_tracker->calc_utilization(seg_addr.get_segment_id());
     [[maybe_unused]] auto ret = space_tracker->allocate(
       seg_addr.get_segment_id(),
       seg_addr.get_segment_off(),
       len);
+    auto new_usage = space_tracker->calc_utilization(seg_addr.get_segment_id());
+    stats.segment_util.buckets[
+      old_usage > 0.5 ?
+	std::lround(old_usage * 10 - 0.5) :
+	std::lround(old_usage * 10)].count--;
+    stats.segment_util.buckets[
+      new_usage > 0.5 ?
+	std::lround(new_usage * 10 - 0.5) :
+	std::lround(new_usage * 10)].count++;
```

Comment:
What is this rounding supposed to do?

Reply:
The segment_util is splitted into several areas, each representing 10% percent, this rounding is to make same 29% usage fall in the 20% percent area.

Reply:
The math looks wrong to me, particularly the 'new_usage > 0.5' bit.  I think you actually want std::lround(std::floor(new_usage * 10)).  Also, this logic is replicated several times in this file, please add a 'long usage_to_histogram_bin(double)' helper.